packages/Confirmation/src/Confirmation.js

```diff
@@ -73,29 +75,26 @@ const Confirmation = props => {
     popoverKey = uuid();
   }, [isPending]);
 
-  const triggerProps = {
-    isConfirmOpen,
-    handleOpenConfirm,
-  };
-
   const handleCloseConfirm = () => {
-    if (isConfirmOpen) {
-      setIsConfirmOpen(false);
-      if (triggerRef.current) triggerRef.current.focus();
-      setTimeout(onClose, 250);
-    }
+    setIsConfirmOpen(prevIsConfirmOpen => {
+      if (prevIsConfirmOpen) {
+        setTimeout(onClose, 250);
+      }
+      return false;
+    });
```

Comment:
Due to confirm and cancel button clicks calling this function as well as popover `onClose` event this is being called twice, and both time `isConfirmOpen` value in state is true, even though it is being set as `false` in the `handleCloseConfirm` function. By checking the previous state of `isConfirmOpen` instead (which is available from the `setIsConfirmOpen` function we are able to get the correct value and contents of the `if` are evaluated only once.